File: 1013-partition-array-into-three-parts-with-equal-sum/1013-partition-array-into-three-parts-with-equal-sum.cpp
class Solution {
public:
    bool canThreePartsEqualSum(vector<int>& arr) {
        int tot=0;
        int zero=0;
        int neg=0;
        int pos=0;
        for(int x:arr)
        {
            tot+=x;
           
        }
        
        if(tot%3!=0)
        {
            return false;
        }
        
        int div=tot/3;
        int checksum=0;
        int count=0;
        for(int i=0;i<arr.size();i++)
        {
            checksum+=arr[i];
            if(checksum==div)
            {
                checksum=0;
                count++;
            }
        }
        return count>=3;
    }
};

File: 1013-partition-array-into-three-parts-with-equal-sum/1013-partition-array-into-three-parts-with-equal-sum.java
class Solution {
    public boolean canThreePartsEqualSum(int[] arr) {
        int tot=0;
        for(var x:arr){
            tot+=x;
        }
        if(tot%3!=0) return false; 
        int part=tot/3;
        int sum1=0,sum2=0;
        boolean aa=false;
        int i=0;
        for(i=0;i<arr.length;i++){
            if(!aa){
                sum1+=arr[i];
            if(sum1==part){
                aa=true;
            }
            }
            else{
                sum2+=arr[i];
                if(sum2==part){
                    break;
                }
            }
        }
        return sum1==sum2 && i<arr.length-1;
    }
}